### DISCUSSION TITLE
Mirth Client - OpenJDK compatibility

### DISCUSSION THREAD

**Discussion by alfonrodrisimon** [nextgenhealthcare/connect#4997] [Q&A]
Has somebody been able to make Mirth Client work with OpenJDK? I have tried with OpenWebstart and IcedTea Web and I cannot make it work. I get into the login page but there is always an error with the connection to the server.  Seems I am not the only one (https://github.com/karakun/OpenWebStart/issues/481)

**Comment by jonbartels** [comment]
I filed https://github.com/karakun/OpenWebStart/issues/481

A user in Slack I think did get OWS to work but I cannot find his notes.

https://github.com/nextgenhealthcare/connect/discussions/4984 is relevant I think. Even though this is Mac focused, using Azul **with OpenJFX** seems to work for MC.

What OpenJDK flavor and version are you using? What errors are you getting, if any?

Note that using the Mirth Connect Admin Launcher is the current best practice. Being able to use OWS would be awesome but MCAL is known to work consistently with MC.

**Reply by alfonrodrisimon** [reply]
I have tried several combinations but it would be nice if it will work with latest OpenWebStart 1.5.1 out of the box. With all distributions I have same issue, Mirth Client loads perfectly but when I tried to login I get the next error "There was an error connecting to the server...." I have read that it might be an issue with TLS handshake but I couldn't get any log from the server. 

The idea of using OpenWebStart comes from customers in order to avoid Java and the new licensing model. Not sure if MCAL applies or not. I know Java is bundled but not sure if it implies a license.

**Reply by pacmano1** [reply]
No paid license is needed for MCAL at all.

**Reply by alfonrodrisimon** [reply]
Can you specify why not? To give a proper answer to customers.

**Reply by pacmano1** [reply]
Nextgen addressed that the Java version embedded did not fall under Oracles paid  license terms on a developer call.  If you need an official answer, you should reach out to Nextgen.

**Comment by trourance** [comment]
> Has somebody been able to make Mirth Client work with OpenJDK? I have tried with OpenWebstart and IcedTea Web and I cannot make it work. I get into the login page but there is always an error with the connection to the server. Seems I am not the only one ([karakun/OpenWebStart#481](https://github.com/karakun/OpenWebStart/issues/481))

I had great succes running it with AdoptOpenJDX 11 LTS with both JVMs in the past

**Reply by pacmano1** [reply]
The recommended way to launch the mirth admin client is with the admin launcher downloadable at https://www.nextgen.com/products-and-services/mirth-connect-downloads.   That tool has a bundled JRE (v8).

I have tried messing with IcedTea and others to launch the jnlp, no luck either.
